File: appjava/src/main/java/cn/ijero/appjava/MainRecyclerAdapter.java
package cn.ijero.appjava;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

/**
 * @author Jero . Created on 2018/1/19.
 */

public class MainRecyclerAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_home_recycler, parent, false);
        return new ViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
        if (holder instanceof ViewHolder) {
            ((ViewHolder) holder).notifyUI();
        }
    }

    @Override
    public int getItemCount() {
        return 40;
    }

    private class ViewHolder extends RecyclerView.ViewHolder {

        ViewHolder(View itemView) {
            super(itemView);
        }

        void notifyUI() {
            ((TextView) itemView.findViewById(R.id.itemTextView)).setText("测试条目 $adapterPosition 测试条目 测试条目 测试条目 测试条目 测试条目 测试条目");

        }
    }

}

File: app/src/main/java/cn/ijero/bhlayout/MainRecyclerAdapter.kt
package cn.ijero.bhlayout

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.item_home_recycler.view.*

/**
 * Created by Administrator on 2017/11/22.
 */
class MainRecyclerAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    override fun onBindViewHolder(holder: RecyclerView.ViewHolder?, position: Int) {
        if (holder is ViewHolder){
            holder.notifyUI()
        }
    }

    override fun getItemCount() = 40

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_home_recycler, parent, false))
    }

    private inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun notifyUI() {
            itemView.itemTextView.text = "测试条目 $adapterPosition 测试条目 测试条目 测试条目 测试条目 测试条目 测试条目"
        }
    }
}